app/modules.go

```diff
@@ -145,53 +145,54 @@ func appModules(
+	// // requires more thought into if it breaks existing logic.
+	// ord.FirstElements(upgradetypes.ModuleName, epochstypes.ModuleName, capabilitytypes.ModuleName)
+
+	// // IBC ordering constraint
+	// ord.Sequence(ibchost.ModuleName)
+	// // token distribution ordering constraint
+	// // staking, slashing & superfluid should come after distribution
+	// ord.Before(distrtypes.ModuleName, slashingtypes.ModuleName)
+	// ord.Before(distrtypes.ModuleName, stakingtypes.ModuleName)
+	// ord.Before(distrtypes.ModuleName, superfluidtypes.ModuleName)
+	// // evidence should come after slashing
+	// ord.After(evidencetypes.ModuleName, slashingtypes.ModuleName)
 	return []string{
 		// Upgrades should be run VERY first
 		upgradetypes.ModuleName,
 		// Note: epochs' begin should be "real" start of epochs, we keep epochs beginblock at the beginning
 		epochstypes.ModuleName,
 		capabilitytypes.ModuleName,
-		minttypes.ModuleName,
-		poolincentivestypes.ModuleName,
+		minttypes.ModuleName,           // no begin block logic
+		poolincentivestypes.ModuleName, // no begin block logic
 		distrtypes.ModuleName,
 		slashingtypes.ModuleName,
 		evidencetypes.ModuleName,
 		stakingtypes.ModuleName,
-		ibchost.ModuleName,
-		ibctransfertypes.ModuleName,
-		icatypes.ModuleName,
-		authtypes.ModuleName,
-		banktypes.ModuleName,
-		govtypes.ModuleName,
-		crisistypes.ModuleName,
-		genutiltypes.ModuleName,
-		authz.ModuleName,
-		paramstypes.ModuleName,
-		vestingtypes.ModuleName,
-		gammtypes.ModuleName,
-		incentivestypes.ModuleName,
-		lockuptypes.ModuleName,
-		poolincentivestypes.ModuleName,
-		tokenfactorytypes.ModuleName,
+		ibchost.ModuleName,             // no begin block logic
+		ibctransfertypes.ModuleName,    // no begin block logic
+		icatypes.ModuleName,            // no begin block logic
+		authtypes.ModuleName,           // no begin block logic
+		banktypes.ModuleName,           // no begin block logic
+		govtypes.ModuleName,            // no begin block logic (only end block oddly)
+		crisistypes.ModuleName,         // no begin block logic (only end block)
+		genutiltypes.ModuleName,        // no begin block logic
+		authz.ModuleName,               // no begin block logic
+		paramstypes.ModuleName,         // no begin block logic
+		vestingtypes.ModuleName,        // no begin block logic
+		gammtypes.ModuleName,           // no begin block logic
+		incentivestypes.ModuleName,     // no begin block logic
+		lockuptypes.ModuleName,         // no begin block logic
+		poolincentivestypes.ModuleName, // no begin block logic
+		tokenfactorytypes.ModuleName,   // no begin block logic
 		// superfluid must come after distribution and epochs
 		superfluidtypes.ModuleName,
-		bech32ibctypes.ModuleName,
-		txfeestypes.ModuleName,
-		wasm.ModuleName,
+		bech32ibctypes.ModuleName, // no begin block logic
+		txfeestypes.ModuleName,    // no begin block logic
+		wasm.ModuleName,           // no begin block logic
 	}
```

Comment:
Please verify this no-op list. 

Links to modules that arent in the SDK or Osmosis:
* ibchost.ModuleName - Actually does have a begin block! Added back in subsequent commit. Proof of name https://github.com/cosmos/ibc-go/blob/main/modules/core/module.go#L43-L46 (I made an issue in ibc-go for this being an odd location for the name), BeginBlock https://github.com/cosmos/ibc-go/blob/main/modules/core/module.go#L165-L168
* ibctransfertypes.ModuleName - https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/module.go#L144-L146
* icatypes.ModuleName - https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/module.go#L191
* wasm: https://github.com/CosmWasm/wasmd/blob/main/x/wasm/module.go#L176

Reply:
No-op list LGTM